resources/views/partials/components.blade.php

```diff
@@ -29,3 +47,24 @@
     @endforeach
 </ul>
 @endif
+
+<script>
+(function () {
```

Comment:
Really, I don't think we need any JS for the dropdown. You can use Blade to find which group is selected and pass that through to the view. This would make the groups unusable for somebody without JS, which isn't needed.